### DISCUSSION TITLE
Too many open socket on Network Extension

### DISCUSSION THREAD

**Discussion by SenberHu** [mitmproxy/mitmproxy#7415] [Q&A]
Mitmproxy: 11.0.2 binary
Python:    3.12.7
OpenSSL:   OpenSSL 3.4.0 22 Oct 2024
Platform:  macOS-15.1-arm64-arm-64bit

I started mimtproxy with mitmproxy --mode local:"Google Chrome", and after a few hours, my Chrome couldn't access the network, I saw the extension error "open too many files" on the Mac console. I execute `lsof -p {org.mitmproxy.macos-redirector.network-extension}`, and saw that the open fd has reached the maximum limit, so when a new flow arrives in the extension, it can't be send to mitmproxy_rs via unix socket, please let me know if there is a problem with my setup or a bug in the program? Looking forward to your reply
<img width="1133" alt="image" src="https://github.com/user-attachments/assets/e316407f-b500-49fe-8763-218d787aa576" />

**Comment by mhils** [comment]
Thank you for the detailed and helpful report, I've converted this into an issue in #7416! :)
